Tests/GraphIR/example.gir

```diff
@@ -0,0 +1,29 @@
+module example where
+
+-- data example.Nat : Type where
+--   example.Nat.z : example.Nat
+--   example.Nat.s : example.Nat -> example.Nat
+
+-- data example.Bool : Type where
+--   example.Bool.true example.Bool.false : example.Bool
+
+-- example.equal : example.Nat -> example.Nat -> example.Bool
+-- example.equal example.Nat.z example.Nat.z = example.Bool.true
+-- example.equal example.Nat.z _ = example.Bool.false
+-- example.equal _ example.Nat.z = example.Bool.false
+-- example.equal (example.Nat.s n) (example.Nat.s m) = example.equal n m
+
+@example : (example.Nat) -> (example.Nat -> _) {
+bb0(%0 : example.Nat, %return: (example.Nat) -> _):
+  %1 = function_ref @example.equal
+  %2 = data_init_simple example.Nat.z : example.Nat
+  apply %1(%0, %2, %bb1) : (example.Bool)
```

Comment:
Note to self: Direct reference to “BB-like” continuations shouldn’t be possible. This should be loaded via `function_ref` and applied. (Also, the apply’s type is incorrect).

Reply:
Speaking of, what's the reasoning for `function_ref` being necessary here?

Reply:
Also: should an `apply` even have a type? It's calling a continuation, so isn't it implicitly `_`?

Reply:
The type annotation on the apply will induce a substitution that should be made explicit.

Also, the `function_ref` is needed because multiple situations now and in the future will induce callable values that should be loaded first. When we emit LLVM IR we can “inline” known `function_ref`s (and eventually `partial_apply`s) as calls.